crates/query-engine/translation/src/translation/helpers.rs

```diff
@@ -89,13 +90,17 @@ pub enum CollectionInfo<'env> {
 }
 
 #[derive(Debug)]
-/// Metadata information about a specific collection.
-pub enum CompositeTypeInfo<'env> {
+/// Metadata information about a specific collection or composite type.
+pub enum CollectionOrCompositeTypeInfo<'env> {
```

Comment:
The reason I called it `CompositeTypeInfo` is that the fact that it represents a type is the interesting thing. Whether this type comes from a collection or a separate composite type in the metadata is only interesting as an implementation detail to lazily query the columns of the type.

Reply:
I agree in theory, but in practice they are represented a bit differently in data.

Reply:
Oh, do you mean whether a type originates from a table vs a distinct composite type has bearing on how its type representations are processed? (I haven't read through the PR completely yet)

Reply:
I'll let you finish and get your feedback :)